internal/services/batch/batch_pool_data_source.go

```diff
@@ -722,154 +722,167 @@ func dataSourceBatchPoolRead(d *pluginsdk.ResourceData, meta interface{}) error
+							}
 
-						for i := 0; i < n; i++ {
-							if v, ok := d.GetOk(fmt.Sprintf("extensions.%d.name", i)); ok && v == *item.Name {
-								extension["protected_settings"] = d.Get(fmt.Sprintf("extensions.%d.protected_settings", i))
-								break
+							for i := 0; i < n; i++ {
+								if v, ok := d.GetOk(fmt.Sprintf("extensions.%d.name", i)); ok && v == item.Name {
+									extension["protected_settings"] = d.Get(fmt.Sprintf("extensions.%d.protected_settings", i))
+									break
+								}
 							}
-						}
 
-						if item.ProvisionAfterExtensions != nil {
-							extension["provision_after_extensions"] = *item.ProvisionAfterExtensions
+							if item.ProvisionAfterExtensions != nil {
+								extension["provision_after_extensions"] = *item.ProvisionAfterExtensions
+							}
+							extensions = append(extensions, extension)
 						}
-						extensions = append(extensions, extension)
+						d.Set("extensions", extensions)
 					}
-					d.Set("extensions", extensions)
-				}
-				if config.ImageReference != nil {
-					d.Set("storage_image_reference", flattenBatchPoolImageReference(config.ImageReference))
-				}
-				if config.LicenseType != nil {
-					d.Set("license_type", *config.LicenseType)
-				}
-				if config.NodeAgentSkuID != nil {
-					d.Set("node_agent_sku_id", *config.NodeAgentSkuID)
-				}
-				if config.NodePlacementConfiguration != nil {
-					nodePlacementConfiguration := make([]interface{}, 0)
-					nodePlacementConfig := make(map[string]interface{})
-					nodePlacementConfig["policy"] = string(config.NodePlacementConfiguration.Policy)
-					nodePlacementConfiguration = append(nodePlacementConfiguration, nodePlacementConfig)
-					d.Set("node_placement", nodePlacementConfiguration)
-				}
-				if config.OsDisk != nil && config.OsDisk.EphemeralOSDiskSettings != nil {
-					d.Set("os_disk_placement", string(config.OsDisk.EphemeralOSDiskSettings.Placement))
-				}
-				if config.WindowsConfiguration != nil {
-					windowsConfig := []interface{}{
-						map[string]interface{}{
-							"enable_automatic_updates": *config.WindowsConfiguration.EnableAutomaticUpdates,
-						},
+
+					d.Set("storage_image_reference", flattenBatchPoolImageReference(&config.ImageReference))
+
+					if config.LicenseType != nil {
+						d.Set("license_type", *config.LicenseType)
+					}
+
+					d.Set("node_agent_sku_id", config.NodeAgentSkuId)
+
+					if config.NodePlacementConfiguration != nil {
+						nodePlacementConfiguration := make([]interface{}, 0)
+						nodePlacementConfig := make(map[string]interface{})
+						nodePlacementConfig["policy"] = string(*config.NodePlacementConfiguration.Policy)
+						nodePlacementConfiguration = append(nodePlacementConfiguration, nodePlacementConfig)
+						d.Set("node_placement", nodePlacementConfiguration)
+					}
+					if config.OsDisk != nil && config.OsDisk.EphemeralOSDiskSettings != nil {
+						d.Set("os_disk_placement", string(*config.OsDisk.EphemeralOSDiskSettings.Placement))
```

Comment:
Does this need to be nil checked too?

```suggestion
					if config.OsDisk != nil && config.OsDisk.EphemeralOSDiskSettings != nil && config.OsDisk.EphemeralOSDiskSettings.Placement != nil{
						d.Set("os_disk_placement", string(*config.OsDisk.EphemeralOSDiskSettings.Placement))
```

Reply:
Thanks @catriona-m!